provider.go

```diff
@@ -74,15 +166,28 @@ func getAPI(hostAddr string, maxRetries int) (client.API, error) {
 func Provider() terraform.ResourceProvider {
 	return &schema.Provider{
 		Schema: map[string]*schema.Schema{
-			"tunnel_address": &schema.Schema{
+			"driver": &schema.Schema{
 				Type:     schema.TypeString,
-				Required: true,
+				Optional: true,
+				Default: "tunnel",
+				Description: "Driver to use to connect to Fleet. Can be tunnel or api.",
```

Comment:
This seems a bit at odds with the capabilities of `fleetctl`, where the `--driver` option can be "API" or "etcd".

`--tunnel` is a different option, and it's possible to combine with `--endpoint`, for instance if you want to tunnel to some host and then access the endpoint there at something other than the default `unix:///var/run/fleet.sock`. I think it would be good to provide a similar functionality here, and keep the configuration item names consistent with the `fleetctl` option names.